linera-core/src/local_node.rs

```diff
@@ -186,50 +211,62 @@ where
     where
         A: LocalValidatorNode + Clone + 'static,
     {
-        let mut info = None;
-        for certificate in certificates {
+        for certificate in &certificates {
             let hash = certificate.hash();
             if !certificate.value().is_confirmed() || certificate.value().chain_id() != chain_id {
                 // The certificate is not as expected. Give up.
                 tracing::warn!("Failed to process network certificate {}", hash);
-                return info;
+                return None;
             }
-            let mut result = self.handle_certificate(certificate.clone(), vec![]).await;
-            if let Err(LocalNodeError::WorkerError(WorkerError::ApplicationBytecodesNotFound(
-                locations,
-            ))) = &result
-            {
-                let chain_id = certificate.value().chain_id();
-                let mut blobs = Vec::new();
-                let maybe_blobs = future::join_all(locations.iter().map(|location| {
+        }
+        let mut result = self.handle_certificates(certificates.clone(), &[]).await;
+        if let Err(LocalNodeError::WorkerError(WorkerError::ApplicationBytecodesNotFound(
+            locations,
+        ))) = &result
+        {
+            let blobs = Self::load_missing_bytecodes(name, node, chain_id, locations).await;
+            result = self.handle_certificates(certificates, &blobs).await;
```

Comment:
This assumes that if one _or more_ of the certificates are missing bytecode, we get _only one_ `ApplicationBytecodesNotFound` error with _all_ the missing bytecode locations. I don't think that's currently the case?